registry/auth/token/accesscontroller.go

```diff
@@ -102,13 +103,37 @@ func (ac authChallenge) Status() int {
 	return http.StatusUnauthorized
 }
 
+func buildAutoRedirectURL(r *http.Request) string {
+	var (
+		scheme = "http"
+		host   = r.Host
```

Comment:
I don't usually pick on code and it might just be that I haven't coded for couple of months now, but:
`host` var doesn't seem to be needed as it's only ever used to init a field in the `u` (`url.URL`) struct; if we are taking this approach why aren't we creating a `path` variable and initializing it to `"/auth/token"` for _consistency sake_? Otherwise, I wouldn't bother creating the `host` var and simply set the `u`'s `Host` field to `r.Host`

Reply:
ok